openff/benchmark/torsiondrives/compute.py

```diff
@@ -0,0 +1,380 @@
+                    task_results[gridpoint] = list()
+
+                    # each gridpoint may have more than one job
+                    for job in next_jobs[gridpoint]:
+                        print(f"... '{gridpoint}'")
+
+                        # set geometry of molecule to that of next job
+                        qcmol_s = qcmol.copy(deep=True).dict()
+                        qcmol_s['geometry'] = np.array(job).reshape(len(qcmol_s['symbols']), 3)
+                        qcmol_s = Molecule.from_data(qcmol_s)
+
+                        # perform optimization
+                        input_data = self._generate_optimization_input(qcmol_s, compute_spec)
+                        result = self._execute_qcengine(input_data,
+                                                        local_options=local_options,
+                                                        scf_maxiter=scf_maxiter,
+                                                        geometric_maxiter=geometric_maxiter,
+                                                        geometric_coordsys=geometric_coordsys,
+                                                        geometric_qccnv=geometric_qccnv)
+
+                        # TODO: consider if we need to do multiple optimizations per grid point to
+                        # get robust results?
+                        task_results[gridpoint].append((result.initial_molecule.geometry.flatten().tolist(),
+                                                        result.final_molecule.geometry.flatten().tolist(),
+                                                        result.energies[-1]))
+
+                        opts[spec_name][gridpoint].append(result)
+
+                td_api.update_state(state, task_results)
+
+        # when complete, return output as a JSON-serializable blob
+        return opts
+
+
+    def _execute_qcengine(
+            self, input_data, procedure="geometric", local_options=None,
+            scf_maxiter=None, geometric_maxiter=None, geometric_coordsys=None,
+            geometric_qccnv=None):
+        import qcengine
+
+        # inject reset=True fix, set coordsys to 'dlc'
+        input_data['keywords']['reset'] = True
+        input_data['keywords']['coordsys'] = 'dlc'
+
+        # inject exposed convergence parameters, if specified
+        if scf_maxiter is not None:
+            input_data['input_specification']['keywords']['maxiter'] = scf_maxiter
+        if geometric_maxiter is not None:
+            input_data['keywords']['maxiter'] = geometric_maxiter
+        if geometric_coordsys is not None:
+            input_data['keywords']['coordsys'] = geometric_coordsys
+        if geometric_qccnv is not None:
+            input_data['keywords']['qccnv'] = geometric_qccnv
+
+        return qcengine.compute_procedure(
+                input_data, procedure=procedure, local_options=local_options)
+
+    @staticmethod
+    def _generate_optimization_input(qcmol, compute_spec):
```

Comment:
@dotsdl it looks like the dihedral is not being frozen or set when passed to geometric so we are probably doing a full optimization, this would make sense as the dihedral in the warning always seems to be the same.

Reply:
you will need something like `constraints_dict = {"set": [{"type": "dihedral", "indices": dihedral, "value": angle}]}` for each constrained dihedral and this needs to be passed with the geometric keywords.

Reply:
Ah, thank you!

Reply:
This looks like it did the trick @jthorton! We have what appears a working torsiondrive executor. Thank you so much for spotting this!